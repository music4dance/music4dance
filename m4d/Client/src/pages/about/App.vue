<template>
  <page id="app" title="music4dance background info" :breadcrumbs="breadcrumbs">
    <h2>About music4dance</h2>
    <p>
      <b>music4dance</b> is a project that has been floating around in my head
      in various forms since I first started ballroom dancing - so what
      <i>is</i> music4dance, really?
    </p>
    <ul>
      <li>
        <b>music4dance</b> is meant to help find music to dance to as well has
        help figure out what dances can be danced to a piece of music
      </li>
      <li>
        <b>music4dance</b> intended to help DJs find new music to add to their
        playlists as well as for students and teachers to build playlists to
        practice to
      </li>
      <li>
        <b>music4dance</b> meant to help dancers understand the relationship
        between dance and the music that they dance to
      </li>
      <li>
        <b>music4dance</b> a place for people to find music to dance to for
        special occassions like weddings and holidy parties
      </li>
    </ul>
    <p>
      Even to make a dent in any one of those goals is pretty ambitious. That's
      about all I've done is to make a dent. I certainly haven't solved any of
      these problems to my satisfaction. So most of all music4dance is a work in
      progress and a playground for me to experiment with the intersection of
      three of my lifelong passions - music, dance and programming.
    </p>
    <p>
      And while I'm doing that, I hope that music4dance provides at least a
      partial solution to some of the above for you. I'm sure that's true for
      some and I can say that some of the most interesting features of this site
      have started as questions from the community. So please keep using the
      site and <a href="https://music4dance.blog/feedback/">let me know</a> what
      you think.
    </p>
    <p class="neutral">
      Please read my <a href="https://www.music4dance.blog">blog</a> for more
      about the history and motivation behind
      <a href="https://www.music4dance.net">music4dance</a>.
    </p>
    <h2>About me</h2>
    <p>
      I’m an amateur dancer and with some training in music who also happens to
      be a professional software engineer. I love dancing, and I particularly
      love dancing when I enjoy the music that I’m dancing to.
    </p>
    <p>
      My dance background is primarily in Ballroom. I’ve been an amateur
      competitor in the
      <a href="https://www.music4dance.net/dances/american-rhythm"
        >American Rhythm style</a
      >). Recently I've moved more towards social partner dancing. As I was
      starting to work on music4dance in earnest, I dove into tap.
    </p>
    <p>
      Most of my programming time outside of my main day jobs has been related
      to music.
    </p>
    <ul>
      <li>
        When the Macintosh came out, my first thought was - Music Editor. And I
        wrote one - although I never got it to the point where I could sell it.
      </li>
      <li>
        My work study project in college was writing music theory software for
        the Macintosh
      </li>
      <li>
        When I left Corporate America to try my hand at the startup world I
        joined a company that was digitizing sheet music and did the initial
        work to take that online.
      </li>
      <li>
        When I left Corporate America for the second time to rediscover my love
        of programming (and generally find myself) I founded music4dance
      </li>
    </ul>
    <h2>My Resume</h2>
    <p>
      <a href="/Content/DavidGray-Resume.pdf">Printable Version</a>
    </p>
    <p>
      <a href="/Home/Resume">Web Version</a>
    </p>
  </page>
</template>

<script lang="ts">
import Page from "@/components/Page.vue";
import { BreadCrumbItem, infoTrail } from "@/model/BreadCrumbItem";
import { Component, Vue } from "vue-property-decorator";

@Component({
  components: {
    Page,
  },
})
export default class App extends Vue {
  private breadcrumbs: BreadCrumbItem[] = [
    ...infoTrail,
    { text: "About", active: true },
  ];
}
</script>
