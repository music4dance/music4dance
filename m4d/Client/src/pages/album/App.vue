<template>
  <page id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <song-table
      :histories="model.histories"
      :filter="model.filter"
      :hideSort="true"
      :hiddenColumns="hidden"
      @song-selected="selectSong"
    ></song-table>
    <admin-footer :model="model" :selected="selected"></admin-footer>
  </page>
</template>

<script lang="ts">
import Page from "@/components/Page.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { safeEnvironment } from "@/helpers/DanceEnvironmentManager";
import { AlbumModel } from "@/model/AlbumModel";
import { DanceEnvironment } from "@/model/DanceEnvironment";
import "reflect-metadata";
import { TypedJSON } from "typedjson";
import SongSelector from "@/mix-ins/SongSelector";

declare const model: string;

export default SongSelector.extend({
  components: { AdminFooter, Page, SongTable },
  data() {
    return new (class {
      model: AlbumModel = TypedJSON.parse(model, AlbumModel)!;
      environment: DanceEnvironment = safeEnvironment();
    })();
  },
  computed: {
    artistRef(): string {
      return `/song/artist?name=${encodeURIComponent(this.model.artist || "")}`;
    },
    hidden(): string[] {
      return this.model.artist ? ["Artist"] : [];
    },
  },
});
</script>
