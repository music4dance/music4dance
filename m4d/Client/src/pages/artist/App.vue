<template>
  <page id="app" :title="title" :consumesEnvironment="true">
    <song-table 
      :songs="model.songs"
      :filter="model.filter"
      :userName="model.userName"
      :hideSort="true"
      :hiddenColumns="['Artist']"
    ></song-table>
  </page>
</template>

<script lang="ts">
// tslint:disable: max-classes-per-file
import 'reflect-metadata';
import { Component, Vue } from 'vue-property-decorator';
import SongTable from '@/components/SongTable.vue';
import Page from '@/components/Page.vue';
import { TypedJSON } from 'typedjson';
import { Song } from '@/model/Song';
import { ArtistModel } from '@/model/ArtistModel';

declare const model: string;

@Component({
  components: {
    Page,
    SongTable,
  },
})
export default class App extends Vue {
  private readonly model: ArtistModel;

  constructor() {
    super();

    this.model = TypedJSON.parse(model, ArtistModel)!;
  }

  private get title(): string {
    return `Artist: ${this.model.artist}`;
  }
}
</script>
