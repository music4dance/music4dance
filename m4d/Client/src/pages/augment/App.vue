<template>
  <page
    id="app"
    :consumesEnvironment="true"
    @environment-loaded="onEnvironmentLoaded"
  >
    <b-row v-if="phase === 'lookup'">
      <b-col>
        <b-alert v-if="lastSong" dismissible show="10">
          Thank you for {{ created ? "adding" : "editing" }}
          <i>{{ lastSong.title }}</i> by {{ lastSong.artist }}
        </b-alert>
        <h1>Add Song</h1>
        <b-tabs v-if="canAugment" v-model="tabIndex" card>
          <b-tab title="by Title"
            ><b-card-text
              ><augment-search
                @edit-song="editSong($event)"
              ></augment-search></b-card-text
          ></b-tab>
          <b-tab title="by Id">
            <b-card-text
              ><augment-lookup @edit-song="editSong($event)"></augment-lookup
            ></b-card-text>
          </b-tab>
          <b-tab v-if="isAdmin" title="Admin">
            <b-card-text>
              <p>Paste in text version of propery list</p>
              <b-input-group prepend="Properties">
                <b-form-input
                  id="admin-properties"
                  palceholder="Song Properties TSV"
                  v-model="propertiesString"
                  trim
                ></b-form-input>
                <b-button variant="primary" @click="adminCreate"
                  >Create</b-button
                >
              </b-input-group>
            </b-card-text>
          </b-tab>
        </b-tabs>
        <augment-info v-else> </augment-info>
      </b-col>
    </b-row>
    <div v-else>
      <b-alert show variant="success">
        <div v-if="songModel.created">
          <b>Create Song:</b> This song is new to music4dance, please fill in
          missing fields and click <b>Add Song</b> to add this song to the
          catalog. Remember, you must vote on at least one dance to add this
          song.
        </div>
        <div v-else>
          <b>Edit Song:</b> We found this song in the music4dance catalog,
          please vote on dance styles and add tags to improve the catalog.
        </div>
      </b-alert>
      <song-core
        :model="songModel"
        :environment="environment"
        :startEditing="true"
        :creating="songModel.created"
        @song-saved="reset(true)"
        @cancel-changes="reset(false)"
      ></song-core>
    </div>
  </page>
</template>

<script lang="ts">
import "reflect-metadata";
import { Component, Mixins } from "vue-property-decorator";
import AdminTools from "@/mix-ins/AdminTools";
import AugmentInfo from "./components/AugmentInfo.vue";
import AugmentLookup from "./components/AugmentLookup.vue";
import AugmentSearch from "./components/AugmentSearch.vue";
import SongCore from "@/pages/song/components/SongCore.vue";
import Page from "@/components/Page.vue";
import { SongDetailsModel } from "@/model/SongDetailsModel";
import { SongFilter } from "@/model/SongFilter";
import { DanceEnvironment } from "@/model/DanceEnvironmet";
import { Song } from "@/model/Song";
import { SongHistory } from "@/model/SongHistory";

enum AugmentPhase {
  lookup = "lookup",
  results = "results",
  edit = "edit",
}

@Component({
  components: { AugmentInfo, AugmentLookup, AugmentSearch, SongCore, Page },
})
export default class App extends Mixins(AdminTools) {
  private phase = AugmentPhase.lookup;
  private songModel: SongDetailsModel | null = null;
  private lastSong: Song | null = null;
  private created = false;
  private environment: DanceEnvironment | null = null;
  private propertiesString = "";
  private tabIndex = 0;

  private get canAugment(): boolean {
    return this.canTag || this.isPremium;
  }

  private editSong(model: SongDetailsModel): void {
    this.songModel = new SongDetailsModel({
      created: !!model.created,
      songHistory: model.songHistory,
      filter: new SongFilter(),
      userName: this.userName,
    });
    this.phase = AugmentPhase.edit;
  }

  private async adminCreate(): Promise<void> {
    console.log(`Create Song: ${this.propertiesString}`);
    this.songModel = new SongDetailsModel({
      created: true,
      songHistory: SongHistory.fromString(this.propertiesString),
      filter: new SongFilter(),
      userName: this.userName,
    });
    this.propertiesString = "";
    this.phase = AugmentPhase.edit;
  }

  private reset(saved: boolean): void {
    if (saved) {
      this.lastSong = Song.fromHistory(this.songModel!.songHistory);
      this.created = !!this.songModel!.created;
    }
    this.phase = AugmentPhase.lookup;
    this.songModel = null;
  }

  private onEnvironmentLoaded(environment: DanceEnvironment): void {
    this.environment = environment;
  }
}
</script>

<style lang="scss" scoped>
#service-id {
  width: 25rem;
}
</style>
