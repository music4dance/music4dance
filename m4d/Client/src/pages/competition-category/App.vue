<template>
  <page id="app" :title="category.name" :breadcrumbs="breadcrumbs">
    <ballroom-list :name="category.name"></ballroom-list>
    <competition-category-table
      :dances="category.round"
      :title="category.fullRoundTitle"
    ></competition-category-table>
    <competition-category-table
      v-if="category.extras"
      :dances="category.extras"
      :title="category.extraDancesTitle"
    ></competition-category-table>
    <div>
      Other categories of <a :href="groupLink">{{groupTitle}}</a> are:
      <ul>
        <li v-for="category in groupModel.otherCategories" :key="category.name">
          <link-category :name="category.name"></link-category>
        </li>
      </ul>
    </div>
    <dl>
      <tempi-link></tempi-link>
      <blog-tag-link title="Ballroom" tag="ballroom"></blog-tag-link>
    </dl>
  </page>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Page from '@/components/Page.vue';
import BallroomList from '@/components/BallroomList.vue';
import LinkCategory from '@/components/LinkCategory.vue';
import TempiLink from '@/components/TempiLink.vue';
import BlogTagLink from '@/components/BlogTagLink.vue';
import CompetitionCategoryTable from '@/components/CompetitionCategoryTable.vue';
import { CompetitionGroupModel, CompetitionCategory } from '@/model/Competition';
import { TypedJSON } from 'typedjson';
import { BreadCrumbItem, ballroomTrail } from '@/model/BreadCrumbItem';

declare const model: string;

@Component({
  components: {
    BallroomList,
    BlogTagLink,
    CompetitionCategoryTable,
    LinkCategory,
    Page,
    TempiLink,
  },
})
export default class App extends Vue {
  private groupModel: CompetitionGroupModel;
  private breadcrumbs: BreadCrumbItem[];

  constructor() {
    super();

    const serializer = new TypedJSON(CompetitionGroupModel);
    const modelS = JSON.stringify(model);
    const modelT = serializer.parse(modelS);
    if (!modelT) {
      throw new Error('Unable to parse model');
    }
    this.groupModel = modelT;
    this.breadcrumbs = [
      ...ballroomTrail,
      { text: this.category.name, active: true},
    ];
  }

  private get category(): CompetitionCategory {
    return this.groupModel.currentCategory;
  }

  private get groupTitle(): string {
    return `competition ${this.groupModel.group.name.toLowerCase()} dancing`;
  }

  private get groupLink(): string {
    return `/dances/${this.groupModel.group.name.toLowerCase()}-competition-categories`;
  }
}
</script>
