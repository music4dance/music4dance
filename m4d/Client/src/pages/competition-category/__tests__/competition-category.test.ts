import { testPage } from "@/helpers/TestHelpers";
import App from "../App.vue";
import { model } from "./model";

describe("CompetitionCategory.vue", () => {
  test("renders a competition category page", () => {
    testPage(App, model);
  });
});
