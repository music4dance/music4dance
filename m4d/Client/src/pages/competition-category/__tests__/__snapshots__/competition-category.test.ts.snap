// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetitionCategory.vue renders a competition category page 1`] = `
<div>
  <div>
    <!---->
    <!---->
    <nav class="navbar navbar-dark bg-primary navbar-expand-lg" id="mainMenu"><a href="/" target="_self" class="navbar-brand"><img src="/images/header-logo.png" height="40" title="music4dance"></a> <button type="button" aria-label="Toggle navigation" class="navbar-toggler collapsed" aria-expanded="false" aria-controls="nav-collapse"><span class="navbar-toggler-icon"></span></button>
      <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing">
        <div id="nav-collapse" class="navbar-collapse collapse" style="display: none;">
          <ul class="navbar-nav">
            <li id="music-menu" class="nav-item b-nav-dropdown dropdown"><a id="music-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="music-menu__BV_toggle_menu_" href="#music-menu" target="_self" class="nav-link dropdown-toggle"><span>Music</span></a>
              <ul tabindex="-1" aria-labelledby="music-menu__BV_toggle_" id="music-menu__BV_toggle_menu_" class="dropdown-menu">
                <li role="presentation"><a role="menuitem" href="/dances" target="_self" class="dropdown-item">Dances</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/ballroom-competition-categories" target="_self" class="dropdown-item">Ballroom</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/latin" target="_self" class="dropdown-item">Latin</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/swing" target="_self" class="dropdown-item">Swing</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/dances/tango" target="_self" class="dropdown-item">Tango</a></li>
                <li role="presentation"><a role="menuitem" href="/song" target="_self" class="dropdown-item">Song Library</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/augment" target="_self" class="dropdown-item">Add Song</a></li>
                <li role="presentation" class="nav-subitem"><a role="menuitem" href="/song/newmusic" target="_self" class="dropdown-item">New Music</a></li>
                <li role="presentation"><a role="menuitem" href="/dances/wedding-music" target="_self" class="dropdown-item">Wedding</a></li>
                <li role="presentation"><a role="menuitem" href="/song/holidaymusic" target="_self" class="dropdown-item">Holiday</a></li>
                <li role="presentation"><a role="menuitem" href="/song/addtags/?tags=%2BHalloween%3AOther" target="_self" class="dropdown-item">Halloween</a></li>
                <li role="presentation"><a role="menuitem" href="/tag" target="_self" class="dropdown-item">Tags</a></li>
              </ul>
            </li>
            <li id="tools-menu" class="nav-item b-nav-dropdown dropdown"><a id="tools-menu__BV_toggle_" role="button" aria-haspopup="true" aria-expanded="false" aria-controls="tools-menu__BV_toggle_menu_" href="#tools-menu" target="_self" class="nav-link dropdown-toggle"><span>Tools</span></a>
              <ul tabindex="-1" aria-labelledby="tools-menu__BV_toggle_" id="tools-menu__BV_toggle_menu_" class="dropdown-menu">
                <li role="presentation"><a role="menuitem" href="/home/counter" target="_self" class="dropdown-item">Tempo Counter</a></li>
                <li role="presentation"><a role="menuitem" href="/home/tempi" target="_self" class="dropdown-item">Tempi (Tempos)</a></li>
                <li role="presentation"><a role="menuitem" href="/song/advancedsearchform" target="_self" class="dropdown-item">Advanced Search</a></li>
                <!---->
              </ul>
            </li>
            <li class="nav-item b-nav-dropdown dropdown"><a role="button" aria-haspopup="true" aria-expanded="false" href="#" target="_self" class="nav-link dropdown-toggle"><span>Info</span></a>
              <ul tabindex="-1" class="dropdown-menu">
                <li role="presentation"><a role="menuitem" href="https://music4dance.blog/music4dance-help/song-list/" target="_self" class="dropdown-item">Help</a></li>
                <li role="presentation"><a role="menuitem" href="https://music4dance.blog/" target="_self" class="dropdown-item">Blog</a></li>
                <li role="presentation"><a role="menuitem" href="/home/faq" target="_self" class="dropdown-item">FAQ</a></li>
                <li role="presentation"><a role="menuitem" href="/home/about" target="_self" class="dropdown-item">About music4dance</a></li>
                <li role="presentation"><a role="menuitem" href="/home/readinglist" target="_self" class="dropdown-item">Reading List</a></li>
                <li role="presentation"><a role="menuitem" href="/home/sitemap" target="_self" class="dropdown-item">Site Map</a></li>
                <li role="presentation"><a role="menuitem" href="/home/privacypolicy" target="_self" class="dropdown-item">Privacy Policy</a></li>
                <li role="presentation"><a role="menuitem" href="/home/credits" target="_self" class="dropdown-item">Credits</a></li>
              </ul>
            </li>
            <li id="contribute-menu" right="" class="nav-item"><a href="/home/contribute" target="_self" class="nav-link">Contribute</a></li>
            <!---->
          </ul>
          <ul class="navbar-nav ml-auto">
            <li class="form-inline">
              <form class="form-inline"><input type="text" placeholder="Search" class="mr-sm-2 form-control form-control-sm"> <button type="submit" disabled="disabled" class="btn my-2 my-sm-0 btn-secondary btn-sm disabled">Search</button></form>
            </li>
            <li right="" class="nav-item"><a href="/identity/account/register?returnUrl=/" target="_self" class="nav-link">Register</a></li>
            <li right="" class="nav-item"><a href="/identity/account/login?returnUrl=/" target="_self" class="nav-link">Login</a></li>
          </ul>
        </div>
      </transition-stub>
    </nav>
    <!---->
    <form id="logoutForm" action="/identity/account/logout" method="post" style="height: 0px;"><input name="__RequestVerificationToken" type="hidden" value="FOO"> <input type="hidden" name="returnUrl" value="/"> <button id="logout" type="submit" class="btn btn-link"></button></form>
  </div>
  <nav aria-label="breadcrumb">
    <ol class="breadcrumb" style="padding: 0.25rem 0.5rem;">
      <li class="breadcrumb-item"><a href="/" target="_self" class="">music4dance</a></li>
      <li class="breadcrumb-item"><a href="/dances" target="_self" class="">Dances</a></li>
      <li class="breadcrumb-item"><a href="/dances/ballroom-competition-categories" target="_self" class="">Ballroom</a></li>
      <li class="breadcrumb-item active"><span aria-current="location">American Smooth</span></li>
    </ol>
  </nav>
  <div id="body-content" class="container-fluid body-content">
    <h1>American Smooth</h1>
    <div>
      <div>
        <p>
          American Smooth is a category of
          <a href="/dances/ballroom-competition-categories">competition ballroom dancing</a>
          that is defined by the
          <a href="http://www.worlddancesport.org/Rule/Athlete/Competition">World Dance Council</a>
          and
          <a href="https://www.ndca.org/pages/ndca_rule_book/Default.asp">National Dance Council of America</a>.
        </p>
        <ul style="list-style-type: disc;">
          <li>
            Click on any of the tempos in the table below (in the MPM, DanceSport,
            or NDCA columns) to go to a list of songs of that tempo
          </li>
          <li>
            Click on the dance names to go to our details information page for that
            dance style
          </li>
          <li>
            Read our
            <a href="https://music4dance.blog/im-a-competition-ballroom-dancer-can-i-find-practice-songs-that-are-a-specific-tempo/">
              blog post</a>
            for step by step instructions on using our catalog to find music of
            specific style and tempo.
          </li>
          <li><a href="/identity/account/register">Register</a> (it's free) to start
            building your own music lists.
          </li>
        </ul>
      </div>
      <div>
        <h4>An American Smooth four dance round consists of the following:</h4>
        <div class="table-responsive">
          <table role="table" aria-busy="false" aria-colcount="7" class="table b-table table-striped table-hover">
            <!---->
            <!---->
            <thead role="rowgroup" class="">
              <!---->
              <tr role="row" class="">
                <th role="columnheader" scope="col" aria-colindex="1" class="">
                  <div>Name</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="2" class="">
                  <div>MPM</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="3" class="">
                  <div>DanceSport</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="4" class="">
                  <div>NDCA Silver/Gold</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="5" class="">
                  <div>NDCA Bronze</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="6" class="">
                  <div>BPM</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="7" class="">
                  <div>Meter</div>
                </th>
              </tr>
            </thead>
            <tbody role="rowgroup">
              <!---->
              <tr role="row" class="">
                <td aria-colindex="1" role="cell" class=""><a href="slow-waltz">Slow Waltz</a></td>
                <td aria-colindex="2" role="cell" class=""><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90">28-30</a></td>
                <td aria-colindex="3" role="cell" class=""><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90">28-30</a></td>
                <td aria-colindex="4" role="cell" class=""><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=90&amp;tempomax=96">30-32</a></td>
                <td aria-colindex="5" role="cell" class=""><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90">28-30</a></td>
                <td aria-colindex="6" role="cell" class=""><a href="/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90">84-90</a></td>
                <td aria-colindex="7" role="cell" class="">3/4</td>
              </tr>
              <tr role="row" class="">
                <td aria-colindex="1" role="cell" class=""><a href="tango-(ballroom)">Tango (Ballroom)</a></td>
                <td aria-colindex="2" role="cell" class=""><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120">30</a></td>
                <td aria-colindex="3" role="cell" class=""><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120">30</a></td>
                <td aria-colindex="4" role="cell" class=""><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=128">30-32</a></td>
                <td aria-colindex="5" role="cell" class=""><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120">30</a></td>
                <td aria-colindex="6" role="cell" class=""><a href="/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120">120</a></td>
                <td aria-colindex="7" role="cell" class="">4/4</td>
              </tr>
              <tr role="row" class="">
                <td aria-colindex="1" role="cell" class=""><a href="slow-foxtrot">Slow Foxtrot</a></td>
                <td aria-colindex="2" role="cell" class=""><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=136">30-34</a></td>
                <td aria-colindex="3" role="cell" class=""><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128">30-32</a></td>
                <td aria-colindex="4" role="cell" class=""><a href="/song/advancedsearch?dances=SFT&amp;tempomin=128&amp;tempomax=136">32-34</a></td>
                <td aria-colindex="5" role="cell" class=""><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=120">30</a></td>
                <td aria-colindex="6" role="cell" class=""><a href="/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=136">120-136</a></td>
                <td aria-colindex="7" role="cell" class="">4/4</td>
              </tr>
              <tr role="row" class="">
                <td aria-colindex="1" role="cell" class=""><a href="viennese-waltz">Viennese Waltz</a></td>
                <td aria-colindex="2" role="cell" class=""><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162">53-54</a></td>
                <td aria-colindex="3" role="cell" class=""><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162">53-54</a></td>
                <td aria-colindex="4" role="cell" class=""><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=162&amp;tempomax=162">54</a></td>
                <td aria-colindex="5" role="cell" class=""><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162">53-54</a></td>
                <td aria-colindex="6" role="cell" class=""><a href="/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162">159-162</a></td>
                <td aria-colindex="7" role="cell" class="">3/4</td>
              </tr>
              <!---->
              <!---->
            </tbody>
            <!---->
          </table>
        </div>
        <!---->
      </div>
      <div>
        <h4>Other dances categorized as American Smooth, although they aren't part of the competition round, are:</h4>
        <div class="table-responsive">
          <table role="table" aria-busy="false" aria-colcount="7" class="table b-table table-striped table-hover">
            <!---->
            <!---->
            <thead role="rowgroup" class="">
              <!---->
              <tr role="row" class="">
                <th role="columnheader" scope="col" aria-colindex="1" class="">
                  <div>Name</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="2" class="">
                  <div>MPM</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="3" class="">
                  <div>DanceSport</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="4" class="">
                  <div>NDCA Silver/Gold</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="5" class="">
                  <div>NDCA Bronze</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="6" class="">
                  <div>BPM</div>
                </th>
                <th role="columnheader" scope="col" aria-colindex="7" class="">
                  <div>Meter</div>
                </th>
              </tr>
            </thead>
            <tbody role="rowgroup">
              <!---->
              <tr role="row" class="">
                <td aria-colindex="1" role="cell" class=""><a href="peabody">Peabody</a></td>
                <td aria-colindex="2" role="cell" class=""><a href="/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=248">60-62</a></td>
                <td aria-colindex="3" role="cell" class=""><a href="/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=248">60-62</a></td>
                <td aria-colindex="4" role="cell" class=""><a href="/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=248">60-62</a></td>
                <td aria-colindex="5" role="cell" class=""><a href="/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=248">60-62</a></td>
                <td aria-colindex="6" role="cell" class=""><a href="/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=248">240-248</a></td>
                <td aria-colindex="7" role="cell" class="">4/4</td>
              </tr>
              <!---->
              <!---->
            </tbody>
            <!---->
          </table>
        </div>
        <!---->
      </div>
      <div>
        Other categories of <a href="/dances/ballroom-competition-categories">competition ballroom dancing</a> are:
        <ul>
          <li><a href="/dances/international-standard">International Standard</a></li>
          <li><a href="/dances/american-rhythm">American Rhythm</a></li>
          <li><a href="/dances/international-latin">International Latin</a></li>
        </ul>
      </div>
      <dl>
        <div>
          <div><img src="/images/icons/tools.png" alt="tools" width="24" height="24">
            Explore relationships between the meter and tempi of different dance styles and find songs that match.
          </div>
          <div class="ml-3"><a href="https://www.music4dance.net/home/tempi" target="_self" class="btn btn-tools btn-sm" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="play" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-play b-icon bi">
                <g>
                  <path d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"></path>
                </g>
              </svg>&nbsp;Try It!
            </a> <a href="https://music4dance.blog/music4dance-help/dance-tempi/" target="_self" class="btn btn-tools btn-sm" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="file text" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-file-text b-icon bi">
                <g>
                  <path d="M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5zM5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z"></path>
                  <path d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z"></path>
                </g>
              </svg>&nbsp;Documentation
            </a> <a href="https://music4dance.blog/tag/tempo/" target="_self" class="btn btn-tools btn-sm" style="margin-left: 1em; margin-bottom: 0.25em;"><svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil square" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil-square b-icon bi">
                <g>
                  <path d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z"></path>
                  <path fill-rule="evenodd" d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z"></path>
                </g>
              </svg>&nbsp;Blog Posts
            </a></div>
          <!---->
        </div>
        <div>
          <dd><img src="/images/icons/info.png" alt="tools" width="24" height="24">
            Read the <a href="https://music4dance.blog">music4dance blog</a> entries
            about <a href="https://music4dance.blog/tag/ballroom">Ballroom</a>.
          </dd>
        </div>
      </dl>
    </div>
  </div>
  <div id="footer-content">
    <hr>
    <footer>
      <p>
        © 2022 -
        <a href="https://www.music4dance.net">Music4Dance.net</a> -
        <a href="https://www.music4dance.net/home/sitemap">Site Map</a> -
        <a href="https://www.music4dance.net/home/termsofservice">Terms of Service</a>
        -
        <a href="https://www.music4dance.net/home/privacypolicy">Privacy Policy</a>
        - <a href="https://www.music4dance.net/home/credits">Credits</a> -
        <a href="https://music4dance.blog/music4dance-help/song-list/">Help</a>
      </p>
    </footer>
  </div>
  <!---->
</div>
`;
